### ISSUE TITLE
Support Cauldron

### ISSUE THREAD

**Issue by oliverdunk** [oliverdunk/JukeboxAPI#41]
We try to use the `net/md_5/bungee/api/chat/BaseComponent` class which is not available in Cauldron.

**Comment by Moutard3** [comment]
Does this affect Sponge too?

**Comment by oliverdunk** [comment]
It's true that we don't support Sponge, but that's for a different reason, so wouldn't be covered by this issue.

Cauldron is based on Bukkit, so should run MCJukebox. It only doesn't because of some more changes that are needed in our clickable chat links.

**Comment by Moutard3** [comment]
Oh okay! Never used any of those two ^^
So would it be better to detect cauldron and use another class, or would it be possible to use another class that the two have?

**Comment by oliverdunk** [comment]
In theory, I think we should be able to find one that the two share. We can go for the other approach if not, though.

Maybe a better idea is just to get the server version and base our decision on that?

**Comment by Moutard3** [comment]
What do you mean with getting the server version?

Edit: Since cauldron is not updated anymore, I think this issue isn't one.
Edit2: LavaBukkit seems more viable option

**Comment by oliverdunk** [comment]
I think in the modding community versions like 1.7 are more common. I could be wrong though.

With the version, I know that /version tells you if you're running Bukkit or Spigot. I hope there's a method for that and it will presumably identify Cauldron too.

**Comment by Moutard3** [comment]
Yeah but that's not what i mean. Cauldron development is abandoned so I don't think it's still use by many people as there's some alternatives like LavaBukkit (which support Spigots plugins) & Sponge (which don't afaik).
Supporting Cauldron isn't useful imo

**Comment by oliverdunk** [comment]
Hmm, that'd be interesting to investigate. I haven't heard of LavaBukkit but if that's the case MCJukebox should be tested on that too.

**Comment by Moutard3** [comment]
I never tried LavaBukkit, just gave it as it was the only alternative I could find.
If possible, a plugin rewrite for sponge would be better I think (but much more time consuming)

**Comment by GrangerTheDog** [comment]
This should support Cauldron #55
